blockchain/pool.go

```diff
@@ -133,7 +133,7 @@ func (pool *BlockPool) removeTimedoutPeers() {
 			// curRate can be 0 on start
 			if curRate != 0 && curRate < minRecvRate {
 				err := errors.New("peer is not sending us data fast enough")
-				pool.sendError(err, peer.id)
+				go pool.sendError(err, peer.id)
```

Comment:
We shouldn't be creating go-routines like this if we can help it.

In this case, it shouldn't be necessary because the errorsCh should be draining just fine.  So I'm against this modification which only serves to mask bugs.

Reply:
Yeh I'm not thrilled with it but was trying to check if this kind of thing would help since it seemed like theres a deadlock somewhere and this is called within a locked function and sends on a channel. Note we already were doing this for the `redo()` below ...